### ISSUE TITLE
glz::expected implicitly convertible to std::expected

### ISSUE THREAD

**Issue by jbbjarnason** [stephenberry/glaze#229]
For use cases where you already have `std::expected` and would like the user facing code to use the standard library would it be beneficial to make `glz::expected` implicitly convertible to `std::expected`?

example 
```C++
#include <expected>
#include <glaze/glaze.hpp>

std::expected<double, int> value = glz::expected<double, int>{}; 
```

or  use std expected when it is available, something like

```C++

#ifdef __cpp_lib_expected
namespace glz {
template<typename expected_t, typename unexpected_t>
using expected<expected_t, unexpected_t> = std::expected<expected_t, unexpected_t>;

template<typename unexpected_t>
using unexpected<unexpected_t> = std::unexpected<unexpected_t>;
}
#else
// internal impl
#endif

```

I know it is configurable whether `glz::expected` should throw or abort, not sure how that would best integrate with this.

Maybe third option to use std::expected explicitly but not sure it is a good idea.

What is the best way around this matter?

**Comment by mwalcott3** [comment]
I think using std::expected when it is available is preferable, Ill go ahead and make that change later today.

**Comment by mwalcott3** [comment]
Addressed in #235. glz::expected is just an alias to std::expected if it is available.
